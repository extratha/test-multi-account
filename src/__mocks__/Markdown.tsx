const MockMarkdown = (props: any) => {
  return <div {...props} />;
};

jest.mock("react-markdown", () => MockMarkdown);
