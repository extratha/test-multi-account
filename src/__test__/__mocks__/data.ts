export const data = {
  test: 'test'
}