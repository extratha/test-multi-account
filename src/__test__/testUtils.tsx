import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
