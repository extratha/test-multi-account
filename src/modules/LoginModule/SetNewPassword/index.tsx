const SetNewPassword = () => {
  return (
    <>
    SET NEW PASS
    </>
  )
}
export default SetNewPassword;