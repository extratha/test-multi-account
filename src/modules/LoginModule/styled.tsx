import { styled, Container, TextField } from "@mui/material"

export const LoginContainer = styled(Container)(() => ({
    backgroundColor: "#1E6AF6",
    margin: "auto",
    borderRadius: "30px",
    maxWidth: 'unset',
}));

export const CustomTextField = styled(TextField)(({ theme }) => ({
    background: theme.palette.background.paper,
    borderRadius: "8px",
}));