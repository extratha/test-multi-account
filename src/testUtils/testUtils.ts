import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
