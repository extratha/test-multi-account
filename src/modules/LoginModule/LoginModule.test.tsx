import { screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { mockLoginResponse } from "@/__mocks__/data";
import LoginModule from "@/modules/LoginModule";
import { render } from "@/testUtils/testUtils";
import axiosPublicInstance from "@/utils/axios/login";

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    replace: jest.fn(),
  }),
}));

jest.mock("@/utils/axios/login", () => ({
  __esModule: true,
  default: {
    post: jest.fn(),
  },
}));

describe("LoginModule", () => {
  it("to match snap", async () => {
    const { asFragment } = render(<LoginModule />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the login form with email and password fields", async () => {
    render(<LoginModule />);
    expect(screen.getByPlaceholderText("อีเมล")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("รหัสผ่าน")).toBeInTheDocument();
    expect(screen.getByTestId("button-login")).toBeDisabled();
  });

  it("shows validation messages on email and password fields", async () => {
    render(<LoginModule />);

    const emailInput = screen.getByPlaceholderText("อีเมล");
    const passwordInput = screen.getByPlaceholderText("รหัสผ่าน");

    await userEvent.type(emailInput, "invalid-email");
    await userEvent.type(passwordInput, "123");
    expect(await screen.findByText(/อีเมลไม่ถูกต้อง/)).toBeInTheDocument();
    expect(await screen.findByText(/รหัสผ่านต้องมีอย่างน้อย 8 หลัก/)).toBeInTheDocument();
  });

  it("enables submit button when both fields are valid", async () => {
    render(<LoginModule />);

    const emailInput = screen.getByPlaceholderText("อีเมล");
    const passwordInput = screen.getByPlaceholderText("รหัสผ่าน");
    const submitButton = screen.getByTestId("button-login");

    await userEvent.type(emailInput, "test@example.com");
    await userEvent.type(passwordInput, "Test@passw0rd");

    await waitFor(() => {
      expect(submitButton).not.toBeDisabled();
    });
  });

  it("submits the form with valid inputs", async () => {
    render(<LoginModule />);
    const emailInput = screen.getByPlaceholderText("อีเมล");
    const passwordInput = screen.getByPlaceholderText("รหัสผ่าน");
    const submitButton = screen.getByTestId("button-login");

    await userEvent.type(emailInput, "test@example.com");
    await userEvent.type(passwordInput, "Test@passw0rd");

    await waitFor(() => {
      expect(submitButton).not.toBeDisabled();
    });

    await userEvent.click(submitButton);

    expect(screen.getByText("CARIVA PLAYGROUND")).toBeInTheDocument();
  });

  it("switch field password to type text", async () => {
    render(<LoginModule />);

    const buttonToggleShowPassword = screen.getByTestId("button-toggle-show-password");
    const passwordInput = screen.getByPlaceholderText("รหัสผ่าน");

    await userEvent.click(buttonToggleShowPassword);

    expect(passwordInput).toHaveAttribute("type", "text");
  });

  it("submits the form with valid inputs with password changed false", async () => {
    const responseData = mockLoginResponse;
    responseData.user.passwordChanged = false;
    (axiosPublicInstance.post as jest.Mock).mockResolvedValueOnce({ data: responseData });

    render(<LoginModule />);

    const emailInput = screen.getByPlaceholderText("อีเมล");
    const passwordInput = screen.getByPlaceholderText("รหัสผ่าน");
    const submitButton = screen.getByTestId("button-login");

    await userEvent.type(emailInput, "test@example.com");
    await userEvent.type(passwordInput, "Test@passw0rd");
    await userEvent.click(submitButton);

    expect(screen.getByText("CARIVA PLAYGROUND")).toBeInTheDocument();
  });
});
