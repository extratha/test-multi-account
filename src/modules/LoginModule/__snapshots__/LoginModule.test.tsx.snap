// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModule should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-1 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  max-width: 66%;
  margin: auto 0px;
  padding: 16px;
}

.emotion-3 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 32px;
}

.emotion-3>:not(style):not(style) {
  margin: 0;
}

.emotion-3>:not(style)~:not(style) {
  margin-top: 8px;
}

.emotion-4 {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
  color: #607D8B;
}

.emotion-5 {
  margin: 0;
  font-size: 24px;
  font-weight: 700;
  line-height: 30px;
}

.emotion-6 {
  margin: 0;
  font-size: 12px;
  line-height: 1.5;
  color: #B0BEC5;
}

.emotion-7 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.emotion-7>:not(style):not(style) {
  margin: 0;
}

.emotion-7>:not(style)~:not(style) {
  margin-top: 16px;
}

.emotion-8 {
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
  background: #fff;
  border-radius: 8px;
}

.emotion-9 {
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-radius: 8px;
}

.emotion-9.Mui-disabled {
  color: #B0BEC5;
  cursor: default;
}

.emotion-9:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-9:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-9.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #2196F3;
  border-width: 2px;
}

.emotion-9.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-9.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-10 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 12px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-10::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10:focus {
  outline: 0;
}

.emotion-10:invalid {
  box-shadow: none;
}

.emotion-10::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-10.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #B0BEC5;
}

.emotion-10:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-10:-webkit-autofill {
  border-radius: inherit;
}

.emotion-11 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-12 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-13 {
  margin: 0;
  font-size: 12px;
  font-weight: 500;
  line-height: 1.5;
  text-align: left;
  color: #d32f2f;
}

.emotion-16 {
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-radius: 8px;
  padding-right: 14px;
}

.emotion-16.Mui-disabled {
  color: #B0BEC5;
  cursor: default;
}

.emotion-16:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-16:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-16.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #2196F3;
  border-width: 2px;
}

.emotion-16.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-16.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-17 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 12px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-right: 0;
}

.emotion-17::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17:focus {
  outline: 0;
}

.emotion-17:invalid {
  box-shadow: none;
}

.emotion-17::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-17:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-17.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #B0BEC5;
}

.emotion-17:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-17:-webkit-autofill {
  border-radius: inherit;
}

.emotion-18 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 50%;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.54);
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  overflow: visible;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-18::-moz-focus-inner {
  border-style: none;
}

.emotion-18.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-18 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-18:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-18:hover {
    background-color: transparent;
  }
}

.emotion-18.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-19 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-20 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-25 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: #eeeeee;
  outline: 0;
  border: 0;
  margin: 2em 0 0;
  border-radius: 4px;
  padding: 6px 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  font-family: fontFamily;
  font-weight: bold;
  font-size: 14px;
  line-height: 1.5;
  text-transform: uppercase;
  min-width: 64px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  width: 100%;
  height: 52px;
}

.emotion-25::-moz-focus-inner {
  border-style: none;
}

.emotion-25.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-25 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-25:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(33, 150, 243, 0.04);
}

@media (hover: none) {
  .emotion-25:hover {
    background-color: transparent;
  }
}

.emotion-25.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-25:hover {
  background-color: rgb(77, 171, 245);
  opacity: 0.9;
}

.emotion-26 {
  margin: 0;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.5;
  text-transform: none;
  color: rgba(0, 0, 0, 0.87);
  padding-top: 32px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-align: center;
  cursor: pointer;
}

.emotion-27 {
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 16px 0px 32px;
}

.emotion-27>:not(style):not(style) {
  margin: 0;
}

.emotion-27>:not(style)~:not(style) {
  margin-left: 8px;
}

.emotion-27 .MuiTypography-root {
  cursor: pointer;
  min-width: 160px;
  padding: 8px 16px;
  text-align: center;
}

.emotion-27 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-28 {
  margin: 0;
  font-size: 12px;
  font-weight: 500;
  line-height: 1.5;
}

<div
    class="MuiStack-root emotion-0"
  >
    <div
      class="MuiStack-root emotion-1"
    >
      <div
        class="MuiStack-root emotion-2"
      >
        <div
          class="MuiStack-root emotion-3"
        >
          <span
            class="MuiTypography-root MuiTypography-titleBold emotion-4"
          >
            เข้าสู่ระบบ
          </span>
          <span
            class="MuiTypography-root MuiTypography-headerBold emotion-5"
          >
            CARIVA PLAYGROUND
          </span>
          <span
            class="MuiTypography-root MuiTypography-labelExtraSmall emotion-6"
          >
            V 2.3
          </span>
        </div>
        <form>
          <div
            class="MuiStack-root emotion-7"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-8"
              data-testid="email-input"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl emotion-9"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input emotion-10"
                  id="email"
                  name="email"
                  placeholder="อีเมล"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline emotion-11"
                >
                  <legend
                    class="emotion-12"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <span
              class="MuiTypography-root MuiTypography-labelExtraSmallMedium emotion-13"
              data-testid="email-error"
            />
          </div>
          <div
            class="MuiStack-root emotion-7"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-8"
              data-testid="password-input"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd emotion-16"
              >
                <input
                  aria-invalid="false"
                  autocomplete="password"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd emotion-17"
                  id="password"
                  name="password"
                  placeholder="ลืมรหัสผ่าน"
                  type="password"
                  value=""
                />
                <button
                  aria-label="Toggle show password"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-18"
                  data-testid="button-toggle-show-password"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="Hiding password"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-19"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7M2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2m4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root emotion-20"
                  />
                </button>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline emotion-11"
                >
                  <legend
                    class="emotion-12"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <span
              class="MuiTypography-root MuiTypography-labelExtraSmallMedium emotion-13"
              data-testid="password-error"
            />
          </div>
          <span
            class="MuiTypography-root MuiTypography-labelExtraSmallMedium emotion-13"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary emotion-25"
            data-testid="button-login"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            เข้าสู่ระบบ
          </button>
        </form>
        <span
          class="MuiTypography-root MuiTypography-bodySmallMedium emotion-26"
        >
          ลืมรหัสผ่าน
        </span>
      </div>
    </div>
    <div
      class="MuiStack-root emotion-27"
    >
      <a
        href="/public/consents/terms-conditions"
        target="_blank"
      >
        <span
          class="MuiTypography-root MuiTypography-labelExtraSmallMedium emotion-28"
          data-testid="button-term-and-conditions"
        >
          ข้อกำหนดและเงื่อนไข
        </span>
      </a>
      <a
        href="/public/consents/privacy-policies"
        target="_blank"
      >
        <span
          class="MuiTypography-root MuiTypography-labelExtraSmallMedium emotion-28"
          data-testid="button-privacy-policy"
        >
          นโยบายความเป็นส่วนตัว
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
